test/functional/wallet_send.py

```diff
@@ -0,0 +1,321 @@
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, add_to_wallet=True)
+
+        self.log.info("Don't broadcast...")
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, add_to_wallet=False)
+        assert(res["hex"])
+
+        self.log.info("Return PSBT...")
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, psbt=True)
+        assert(res["psbt"])
+
+        self.log.info("Create transaction that spends to address, but don't broadcast...")
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, add_to_wallet=False)
+        # conf_target & estimate_mode can be set as argument or option
+        res1 = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, arg_conf_target=1, arg_estimate_mode="economical", add_to_wallet=False)
+        res2 = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, conf_target=1, estimate_mode="economical", add_to_wallet=False)
+        assert_equal(self.nodes[1].decodepsbt(res1["psbt"])["fee"],
+                     self.nodes[1].decodepsbt(res2["psbt"])["fee"])
+        # but not at the same time
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, arg_conf_target=1, arg_estimate_mode="economical",
+                       conf_target=1, estimate_mode="economical", add_to_wallet=False, expect_error=(-8,"Use either conf_target and estimate_mode or the options dictionary to control fee rate"))
+
+        self.log.info("Create PSBT from watch-only wallet w3, sign with w2...")
+        res = self.test_send(from_wallet=w3, to_wallet=w1, amount=1)
+        res = w2.walletprocesspsbt(res["psbt"])
+        assert res["complete"]
+
+        self.log.info("Create PSBT from wallet w4 with watch-only keys, sign with w2...")
+        self.test_send(from_wallet=w4, to_wallet=w1, amount=1, expect_error=(-4, "Insufficient funds"))
+        res = self.test_send(from_wallet=w4, to_wallet=w1, amount=1, include_watching=True, add_to_wallet=False)
+        res = w2.walletprocesspsbt(res["psbt"])
+        assert res["complete"]
+
+        self.log.info("Create OP_RETURN...")
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1)
+        self.test_send(from_wallet=w0, data="Hello World", expect_error=(-8, "Data must be hexadecimal string (not 'Hello World')"))
+        self.test_send(from_wallet=w0, data="23")
+        res = self.test_send(from_wallet=w3, data="23")
+        res = w2.walletprocesspsbt(res["psbt"])
+        assert res["complete"]
+
+        self.log.info("Set fee rate...")
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, conf_target=2, estimate_mode="duff/b", add_to_wallet=False)
+        fee = self.nodes[1].decodepsbt(res["psbt"])["fee"]
+        assert_fee_amount(fee, Decimal(len(res["hex"]) / 2), Decimal("0.00002"))
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, conf_target=-1, estimate_mode="duff/b",
+                       expect_error=(-3, "Amount out of range"))
+
```

Comment:
16378: missing
```
        # Fee rate of 0.1 satoshi per byte should throw an error
        # TODO: error should say 1.000 sat/b
        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, conf_target=0.1, estimate_mode="sat/b",
                       expect_error=(-4, "Fee rate (0.00000100 BTC/kB) is lower than the minimum fee rate setting (0.00001000 BTC/kB)"))

        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, conf_target=0.000001, estimate_mode="BTC/KB",
                       expect_error=(-4, "Fee rate (0.00000100 BTC/kB) is lower than the minimum fee rate setting (0.00001000 BTC/kB)"))
```
why?

Reply:
Because the code for throwing that error isn't present in Dash Core because the backport that introduces that check (bitcoin#18275) hasn't been introduced yet. bitcoin#18275 has been marked as "dnm probably?" in the backports Google Sheet.

![bitcoin#18275 marked as DNM candidate](https://github.com/dashpay/dash/assets/63189531/5d02174d-0c82-4b89-8b1d-3989253d9d75)

The PR itself should be relatively trivial to merge except that [`nFeeRateNeeded`](https://github.com/bitcoin/bitcoin/pull/10637/files#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8R2785-R2786) (which is part of the [`nFeeRateNeeded > *coin_control.m_feerate`](https://github.com/bitcoin/bitcoin/pull/18275/files#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8R2734) condition of the "minimum fee rate setting" check) is, alongside other things, introduced in bitcoin#10637, which was introduced in Bitcoin in v0.17.

In the backports Google Sheet, it's with the remark "large conflicts - udjin open PR - partialy? done" and bitcoin#10637 was the recipient of [additional fixes](https://github.com/dashpay/dash/pull/5226/commits/175503304873e89fd05039d472f7da2f9b0284c4) in [dash#5226](https://github.com/dashpay/dash/pull/5226).

With the above two in mind, it seemed that the most straightforward thing to do would be to assume bitcoin#18275 as DNM and drop the functional tests that rely on it.

Reply:
bitcoin#16378 has been dropped from this PR, resolved.